lib/spack/spack/solver/concretize.lp

```diff
@@ -554,8 +554,8 @@ variant(Package, Variant) :- variant_condition(ID, Package, Variant),
 
 attr("variant_propagate", Package, Variant, Value, Source) :-
   attr("node", Package),
-  depends_on(Parent, Package),
-  attr("variant_propagate", Parent, Variant, Value, Source),
+  path(Source, Package),
```

Comment:
Leaving a note from an internal discussion on Slack. It's not advisable to use `path` here, since computing that fact is very expensive (clingo has to compute every possible path in every possible DAG).